src/tribler-gui/tribler_gui/widgets/triblertablecontrollers.py

```diff
@@ -132,6 +132,9 @@ def _on_selection_changed(self, selected, deselected):
 class HealthCheckerMixin:
     def check_torrent_health(self, data_item, forced=False):
         # TODO: stop triggering multiple checks over a single infohash by e.g. selection and click signals
+        if 'infohash' not in data_item:
```

Comment:
As I understand it, channels have infohashes as well as regular torrents, but we probably don't want to show health information for channels in this view.

In other similar places, there is a check `if data_item['type'] != REGULAR_TORRENT`. I think we need to use a similar check here as well.

Reply:
Checking infohash is a more reliable approach if we look at it as a "bugfix".

But your suggestion is the more in `triblertablecontrollers' style`, which is more correct if we look at the change as a "triblertablecontrollers fix".

I've implemented your suggestion (still not sure that this is the best version of bugfix).

Reply:
It is indeed more correct to check for the entry type, not for its attributes. In this way, we catch the errors in the application *logic*, though at the cost of the thing crashing for the user.